rasa_nlu/server.py

```diff
@@ -48,74 +46,117 @@ def requires_auth(f):
 
     @wraps(f)
     def decorated(*args, **kwargs):
-        token = request.args.get('token', '')
-        if current_app.data_router.token is None or token == current_app.data_router.token:
+        self = args[0]
+        request = args[1]
+        token = request.args.get('token', [''])[0]
+        if self.data_router.token is None or token == self.data_router.token:
             return f(*args, **kwargs)
         return "unauthorized", 401
 
     return decorated
 
 
-def create_app(config, component_builder=None):
-    rasa_nlu_app = Flask(__name__)
+class RasaNLU(object):
+    """Class representing RASA NLU http server"""
 
-    @rasa_nlu_app.route("/parse", methods=['GET', 'POST'])
+    app = Klein()
+
+    def __init__(self, config, component_builder=None):
+        logging.basicConfig(filename=config['log_file'], level=config['log_level'])
+        logging.captureWarnings(True)
+        logger.info("Configuration: " + config.view())
+
+        logger.debug("Creating a new data router")
+        self.config = config
+        self.data_router = DataRouter(config, component_builder)
+        reactor.suggestThreadPoolSize(config['num_threads'] * 5)
```

Comment:
Unable to recover the where I saw this (maybe I dreamed it) but it is a recommendation from Twisted about how to use the reactor. I will change it if I can't find the related source.